### PR TITLE
themis: fix ENOENT when rust-toolchain.toml doesn't exist

### PR DESCRIPTION
**Pull request by miraclx**
In the absence of the `rust-toolchain.toml` file, themis exits with "No such file or directory".

Since there's no rule enforcing the presence of a toolchain file, we can ignore its absence.

### CODE REVIEW

File: tools/themis/src/utils.rs [hunk 1]
```diff
@@ -22,12 +22,17 @@ pub fn parse_workspace() -> anyhow::Result<Workspace> {
         .cloned()
         .filter(|package| metadata.workspace_members.contains(&package.id))
         .map(|package| {
-            let raw = read_toml(&package.manifest_path)?;
+            let raw = match read_toml(&package.manifest_path)? {
+                Some(raw) => raw,
+                _ => return Err(anyhow::anyhow!("failed to read package manifest")),
```
Comment (akhi3030) [inline tools/themis/src/utils.rs:27]:
```suggestion
                None => return Err(anyhow::anyhow!("failed to read package manifest")),
```
Same as above.

Comment (akhi3030) [inline tools/themis/src/utils.rs:27]: This is a bit unintuitive.  I can get some failures from line 25 and another failure here.  I wonder if we need to use stronger return types in `read_toml()`.

File: tools/themis/src/utils.rs [hunk 2]
```diff
@@ -4,8 +4,8 @@ use cargo_metadata::{camino::Utf8PathBuf, CargoOpt, MetadataCommand};
 
 use super::types::{Package, Workspace};
 
-pub fn read_toml(path: &Utf8PathBuf) -> anyhow::Result<toml::Value> {
-    Ok(toml::from_slice(&fs::read(path)?)?)
+pub fn read_toml(path: &Utf8PathBuf) -> anyhow::Result<Option<toml::Value>> {
```
Comment (akhi3030) [inline tools/themis/src/utils.rs:7]: Look at the users of this function, I wonder if we should introduce a better return type here.  

I think we care about the following three variants:
- everything succeeded and here is the toml value.
- something failed in trying to read the file.
- something failed in try to parse it into a toml.

I wonder if introducing an explicit type that captures the above three variants might make the code more readable.

Comment (miraclx) [reply tools/themis/src/utils.rs:7]: Constrained the error dismissal to ENOENT, anything else will bubble up to a runtime error. At this point, I believe a `Result<Option<T>>` should suffice.

File: tools/themis/src/utils.rs (hunk 1 above)
Comment (miraclx) [reply tools/themis/src/utils.rs]: At this point, line 25 returns all errors but ENOENT. Just needed to make this error report was more distinctive. Should be now.

**Review by akhi3030 — APPROVED** [review]
thanks!

**Pull request merged**